src/powerquery-language-services/documentSymbols.ts

```diff
@@ -12,23 +12,23 @@ import { Ast } from "@microsoft/powerquery-parser/lib/powerquery-parser/language
 
 import * as InspectionUtils from "./inspectionUtils";
 import { DocumentSymbol, SymbolKind, TextDocument } from "./commonTypes";
-import { WorkspaceCache, WorkspaceCacheUtils } from "./workspaceCache";
+import { WorkspaceCacheUtils } from "./workspaceCache";
 
-export function getDocumentSymbols(
+export async function getDocumentSymbols(
     textDocument: TextDocument,
     lexAndParseSettings: PQP.LexSettings & PQP.ParseSettings,
     maintainWorkspaceCache: boolean,
-): DocumentSymbol[] {
-    const cacheItem: WorkspaceCache.ParseCacheItem = WorkspaceCacheUtils.getOrCreateParse(
+): Promise<DocumentSymbol[]> {
+    const maybeTriedParse: PQP.Task.TriedParseTask | undefined = await WorkspaceCacheUtils.getOrCreateParsePromise(
```

Comment:
It's a habit I have to help keep track of things in my head, but I understand it can be a verbosity issue. I think removing it is a discussion we should have before integration with PQO. If it's removed there then I'd also remove it in the other parts of the stack.